### PR TITLE
Fixed incorrectly generated files

### PR DESCRIPTION
**Pull request by DrizztDoUrden**
This is a hotfix for issue described here: https://github.com/ROCm/MIOpen/pull/2722#issuecomment-1961900323

3 things are fixed:
- text db files now contain correct keys
- any updates to sqlite2txt now force regeneration of db.txt files
- text user dbs now have a separate extension

### CODE REVIEW

**Comment by DrizztDoUrden** [comment]
We should add a system perfdb validity test checking if contains a sane amount of values, they are of sane size and some config can be read by the Db class (that one does the most parsing).
I can make such test on Monday morning

**Comment by atamazov** [comment]
Correctness/Performance testing is in progress.

**Comment by atamazov** [comment]
## :cyclone: Correctness/Performance testing results

Preconditions:
- Navi21 (gfx1030/36)
- ROCm 6.0 docker, Ubuntu 20.04.5 LTS
- BUILD_DEV=On, Release build
- branch ddu/fix-sqlite2txt b536a116d vs. develop 81afa6503 "Bump cryptography from 42.0.2 to 42.0.4 in /docs/sphinx"
- 93 known FP32 convolution configs

Tested modes:
- (1) Find Mode (DYNAMIC_HYBRID)
- (2) Immediate mode with user find-db filled by step (1).

Expected results: No performance or correctness regressions. Aux-wall time improvements.

🟡 Results: No regressions but **_no aux-wall time improvements_**. Most likely, the reasons are
- In DEV builds, the most of the aux wall time is spent for building kernels.
- The size of the system perf-db for Navi21 is very small (720k vs. 10-50M for other targets)

I will re-test non-DEV build on gfx906/60 tomorrow.

Verdict: :green_circle: GO!

**Review by junliume — APPROVED** [review]

**Comment by atamazov** [comment]
@DrizztDoUrden Either this or #2722 breaks compatibility with ROCm 5.7.0:
```bash
# VERBOSE=1 make -j$((`nproc`-4)) MIOpenDriver
...
[  0%] Linking CXX executable ../../bin/sqlite2txt
cd /root/github/MLOpen1/build/5.7.0/tools/sqlite2txt && /usr/bin/cmake -E cmake_link_script CMakeFiles/sqlite2txt.dir/link.txt --verbose=1
...
make -f src/CMakeFiles/MIOpen.dir/build.make src/CMakeFiles/MIOpen.dir/depend
...
/opt/rocm/llvm/bin/clang++  -O3 -DNDEBUG -s   CMakeFiles/sqlite2txt.dir/main.cpp.o  -o ../../bin/sqlite2txt  /opt/rocm/lib/libsqlite3.a
ld.lld: error: undefined symbol: dlopen
>>> referenced by sqlite3.c
>>>               sqlite3.c.o:(unixDlOpen) in archive /opt/rocm/lib/libsqlite3.a

ld.lld: error: undefined symbol: dlerror
>>> referenced by sqlite3.c
>>>               sqlite3.c.o:(unixDlError) in archive /opt/rocm/lib/libsqlite3.a
...
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [tools/sqlite2txt/CMakeFiles/sqlite2txt.dir/build.make:85: bin/sqlite2txt] Error 1
...
```

**Comment by atamazov** [comment]
## :cyclone: Performance testing results

Preconditions:
- gfx906/60, perf-db size: 21M (SQLite), 11M (txt)
- ROCm 6.0 docker, Ubuntu 20.04.5 LTS
- BUILD_DEV=Off, Release build
- branch ddu/fix-sqlite2txt b536a116d vs. develop 81afa6503 "Bump cryptography from 42.0.2 to 42.0.4 in /docs/sphinx"
- 93 known FP32 convolution configs
- Pre-filled binary kernel cache, identical for all runs
- Pre-filled user-find-db, identical for all runs

Tested modes:
- (1) Default Find Mode (DYNAMIC_HYBRID)
- (2) Immediate mode

Expected results: No performance or correctness regressions. Aux-wall time improvements.

Results: No regressions. **_Total wall time_** improvement is:
- Find Mode (DYNAMIC_HYBRID): +16% on average (-5% min, +135% max)
- Immediate mode: +155% (2.55X) on average (-13% min, 475% max)
- Detailed logs & csv files are available upon request.

Bottom line:
- :green_circle: GO!
- From the neural networks' point of view, we can expect improvements of **_initial iteration time_**.
- MI100/200 should reveal bigger performance gain as we have **_~4X bigger perf-db_** for these targets.

/cc @junliume @JehandadKhan @DrizztDoUrden
